src/include/interval_set.h

```diff
@@ -46,6 +46,14 @@ class interval_set {
   class iterator : public std::iterator <std::forward_iterator_tag, T>
   {
     public:
+        using Map = C<T, T>;
```

Comment:
where is `C` defined? and there is no need to expose `Map` here.

Reply:
@tchaikov 
`C` is defined on line 33:
`template<typename T, template<typename, typename, typename ...> class C = std::map>`

But I think that it might be redundant here.

Reply:
ahh, i was looking at the nautilus branch. but `Map` is already defined at line 36. i think there is no need to define it again.